Assets/BML/ScriptableObjectCore/Scripts/ScriptableObjectResetOnEnterPlaymode.cs

```diff
@@ -0,0 +1,61 @@
+using System.Collections.Generic;
+using BML.ScriptableObjectCore.Scripts;
+using BML.ScriptableObjectCore.Scripts.Variables;
+using UnityEngine;
+
+#if UNITY_EDITOR
+using BML.ScriptableObjectCore.Scripts.Utils;
+using UnityEditor;
+#endif
+
+namespace BML.ScriptableObjectCore.Scripts
+{
+    public static class ScriptableObjectResetOnEnterPlaymode
+    {
+        private static HashSet<string> _variablesToReset;
+        
+#if UNITY_EDITOR
+        [InitializeOnLoadMethod]
+        private static void Initialize()
+        {
+            _variablesToReset = new HashSet<string> {};
+            
+            EditorApplication.playModeStateChanged += OnPlayModeStateChanged;
+            Debug.Log($"OnProjectLoadedInEditor: ScriptableObjectInitializer");
+            
+        }
+#endif
```

Comment:
Is there a reason these #if statements are separated?

Reply:
no they can be combined